### PR TITLE
Feature/account migration submit tx

### PR DESCRIPTION
**Pull request by piyalbasu**

### CODE REVIEW

File: @shared/api/types.ts [hunk 1]
```diff
@@ -18,6 +18,8 @@ export interface UserInfo {
   publicKey: string;
 }
 
+export type MigratableAccount = Account & { keyIdIndex: number };
```
Comment (piyalbasu) [inline @shared/api/types.ts:21]: when we migrate an account, we need to know where the in the keyIdList this specific account falls so we can seamlessly replace the old account with the new one

File: extension/src/background/helpers/migration.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    try {
+      builtChangeTrustTx.sign(recipientSourceKeys);
+    } catch (e) {
+      console.error(e);
+    }
+
+    // eslint-disable-next-line no-await-in-loop
+    await server.submitTransaction(builtChangeTrustTx);
+
+    if (asset) {
```
Comment (aristidesstaffieri) [inline extension/src/background/helpers/migration.ts:75]: small nit, is this check needed? Seems like the check on like 54/ instantiating an `Asset` would take care of this.

File: extension/src/background/helpers/migration.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    const builtChangeTrustTx = changeTrustTx.setTimeout(180).build();
+
+    try {
+      builtChangeTrustTx.sign(recipientSourceKeys);
+    } catch (e) {
+      console.error(e);
+    }
+
+    // eslint-disable-next-line no-await-in-loop
+    await server.submitTransaction(builtChangeTrustTx);
```
Comment (aristidesstaffieri) [inline extension/src/background/helpers/migration.ts:73]: Should there be a check for a failed change trust transaction before sending a balance?

File: extension/src/background/messageListener/popupMessageListener.ts [hunk 4]
```diff
@@ -1301,6 +1398,192 @@ export const popupMessageListener = (request: Request, sessionStore: Store) => {
...
+        privateKey: newWallet.getSecret(keyIdIndex),
+      };
+
+      // eslint-disable-next-line no-await-in-loop
+      const transaction = await new TransactionBuilder(sourceAccount, {
+        fee,
+        networkPassphrase: Networks.TESTNET,
+      });
+
+      if (!xlmBalance) {
```
Comment (aristidesstaffieri) [inline extension/src/background/messageListener/popupMessageListener.ts:1462]: Should this condition trigger a user facing error so users know why a balance cannot be migrated?

File: extension/src/background/helpers/migration.ts (hunk 2 above)
Comment (piyalbasu) [reply extension/src/background/helpers/migration.ts:75]: Yup - good call. In the next PR I streamline this whole thing, I'll make this fix there

File: extension/src/background/messageListener/popupMessageListener.ts (hunk 4 above)
Comment (piyalbasu) [reply extension/src/background/messageListener/popupMessageListener.ts:1462]: Yup, I'll work this into the UI

**Pull request merged**